2.train-VAE/scripts/nbconverted/beta_vae_training.py

```diff
@@ -8,77 +8,259 @@
+concat_frames = [train_init, test_init]
+train_and_test = pd.concat(concat_frames).reset_index(drop=True)
+train_and_test[["age_category", "sex"]] = train_and_test.age_and_sex.str.split(pat='_',expand=True)
+train_and_test_subbed = train_and_test.filter(gene_list, axis = 1)
+asdf = []
+asdf = pd.DataFrame(asdf)
+metadata = asdf.assign(
+    DepMap_ID = train_and_test.DepMap_ID.astype(str), 
+    age_category = train_and_test.age_category.astype(str), 
+    sex = train_and_test.sex.astype(str), 
+    train_or_test = train_and_test.train_or_test.astype(str)
+    )
+metadata
+
+
+# In[19]:
+
+
+latent_complete = np.array(encoder.predict(train_and_test_subbed)[2])
+latent_df = pd.DataFrame(latent_complete)
+
+
+# In[20]:
+
+
+age_category = metadata.pop("age_category")
+sex = metadata.pop("sex")
+train_test = metadata.pop("train_or_test")
+
+
+# In[21]:
+
+
+# display clustered heatmap of coefficients
+lut_pal = sns.cubehelix_palette(age_category.unique().size, light=.9, dark=.1, reverse=True, start=1, rot=-2)
+put_pal = sns.cubehelix_palette(sex.unique().size)
+mut_pal = sns.color_palette("hls", train_test.unique().size)
+
+lut = dict(zip(age_category.unique(), lut_pal))
+put = dict(zip(sex.unique(), put_pal))
+mut = dict(zip(train_test.unique(), mut_pal))
+
+row_colors1 = age_category.map(lut)
+row_colors2 = sex.map(put)
+row_colors3 = train_test.map(mut)
+
+network_node_colors = pd.DataFrame(row_colors1).join(pd.DataFrame(row_colors2).join(pd.DataFrame(row_colors3)))
+
+sns.set(font_scale=1.4)
+g = sns.clustermap(latent_df, method='ward', figsize=(10, 20), row_colors= network_node_colors, yticklabels=False, dendrogram_ratio=(.1,.04), cbar_pos=(1,.2, 0.02, .6))
+
+
+xx = []
+for label in age_category.unique():
+    x = g.ax_row_dendrogram.bar(0, 0, color=lut[label], label=label, linewidth=0)
+    xx.append(x)
+# add the legend
+legend3 = plt.legend(xx, age_category.unique(), loc="center", title='age category', ncol=2, bbox_to_anchor=(1.3, .55), bbox_transform=gcf().transFigure)
+
+
+yy = []
+for label in sex.unique():
+    y = g.ax_row_dendrogram.bar(0, 0, color=put[label], label=label, linewidth=0)
+    yy.append(y)  
+# add the second legend
+legend4 = plt.legend(yy, sex.unique(), loc="center", title='sex', ncol=3, bbox_to_anchor=(1.3, .5), bbox_transform=gcf().transFigure)
+plt.gca().add_artist(legend3)
+
+
+zz = []
+for label in train_test.unique():
+    z = g.ax_row_dendrogram.bar(0, 0, color=mut[label], label=label, linewidth=0)
+    zz.append(z)
+# add the third legend
+legend5 = plt.legend(zz, train_test.unique(), loc="center", title='train or test', ncol=2, bbox_to_anchor=(1.3, .45), bbox_transform=gcf().transFigure)
+plt.gca().add_artist(legend4)
 
 
-cp_vae.train(x_train=dependency_df, x_test=dependency_df)
+# save the figure
+heat_save_path = pathlib.Path('../1.data-exploration/figures/heatmap.png')
+g.figure.savefig(heat_save_path, dpi=600)
```

Comment:
I don't see legends in this figure that you're adding to the PR?
[
1.data-exploration/figures/heatmap.png](https://github.com/WayScience/gene_dependency_representations/blob/f7f5585a426fc12bc5d346f4fceee72d448a1bb8/1.data-exploration/figures/heatmap.png)

Is this an old file?

Reply:
it was an issue with the save command. its been resolved and wil push it with the other updates